drools-parser/src/main/java/org/drools/parser/DRLVisitorImpl.java

```diff
@@ -83,6 +87,12 @@ private void applyChildrenDescrs(PackageDescr packageDescr, List<BaseDescr> desc
                     functionDescr.setDialect(dialect.getValue());
                 }
                 packageDescr.addFunction(functionDescr);
+            } else if (descr instanceof TypeDeclarationDescr) {
+                packageDescr.addTypeDeclaration((TypeDeclarationDescr) descr);
+            } else if (descr instanceof EntryPointDeclarationDescr) {
+                packageDescr.addEntryPointDeclaration((EntryPointDeclarationDescr) descr);
+            } else if (descr instanceof WindowDeclarationDescr) {
+                packageDescr.addWindowDeclaration((WindowDeclarationDescr) descr);
```

Comment:
This if-else block doesn't looks nice. I'll address in https://issues.redhat.com/browse/DROOLS-7564